src/plugins/vis_type_timeseries/public/application/components/lib/check_if_numeric_metric.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getIndexPatternKey } from '../../../../common/index_patterns_utils';
+import { METRIC_TYPES } from '../../../../common/enums';
+
+import type { Metric, IndexPatternValue } from '../../../../common/types';
+import type { VisFields } from '../../lib/fetch_fields';
+
+// this function checks if metric has numeric value result
+export const checkIfNumericMetric = (
+  metric: Metric,
+  fields: VisFields,
+  indexPattern: IndexPatternValue
+) => {
+  // currently only Top Hit could have not numeric value result
+  if (metric?.type === METRIC_TYPES.TOP_HIT) {
+    const selectedField = fields[getIndexPatternKey(indexPattern)]?.find(
+      ({ name }) => name === metric?.field
+    );
+    return selectedField?.type === 'number';
```

Comment:
'number' -> KBN_FIELD_TYPES.NUMBER

Reply:
yes @DziyanaDzeraviankina can we also make this nit?

Reply:
And add a small unit test